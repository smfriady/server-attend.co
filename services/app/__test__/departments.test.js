if (process.env.NODE_ENV !== "production") {
  require("dotenv").config();
}

const app = require("../app");
const { Department } = require("../models");
const request = require("supertest");

describe("GET /api/v1/web/departments", () => {
  test("GET /api/v1/web/departments - 200 - Departments - Success", async () => {
    const response = await request(app).get("/api/v1/web/departments");

    expect(response.status).toBe(200);
    expect(response.body).toBeInstanceOf(Array);

    response.body.forEach((employee) => {
      expect(employee).toMatchSnapshot({ id: expect.any(Number) });
    });
    response.body.forEach((employee) => {
      expect(employee).toMatchSnapshot({ name: expect.any(String) });
    });
  });

  test("handle error when hit findAll", async () => {
    const expectedError = "Error bos";
    jest.spyOn(Department, "findAll").mockRejectedValue(expectedError);

    const response = await request(app).get("/api/v1/web/departments");

    expect(response.status).toBe(500);
    // expect(response.body.name).toBe("Error bos"); // harus di handle balikan message errornya
  });
});
