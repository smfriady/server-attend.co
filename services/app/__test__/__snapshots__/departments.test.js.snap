// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 1`] = `
{
  "id": Any<Number>,
  "name": "Marketing",
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 2`] = `
{
  "id": Any<Number>,
  "name": "Engineering",
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 3`] = `
{
  "id": Any<Number>,
  "name": "Management",
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 4`] = `
{
  "id": Any<Number>,
  "name": "Finance",
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 5`] = `
{
  "id": Any<Number>,
  "name": "HR",
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 6`] = `
{
  "id": 1,
  "name": Any<String>,
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 7`] = `
{
  "id": 2,
  "name": Any<String>,
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 8`] = `
{
  "id": 3,
  "name": Any<String>,
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 9`] = `
{
  "id": 4,
  "name": Any<String>,
}
`;

exports[`GET /api/v1/web/departments GET /api/v1/web/departments - 200 - Departments - Success 10`] = `
{
  "id": 5,
  "name": Any<String>,
}
`;
